### DISCUSSION TITLE
Moving fastlane to MNF

### DISCUSSION THREAD

**Discussion by steipete** [MobileNativeFoundation/discussions#194] [Proposals]
fastlane is currently being ignored by Google, yet still on their GitHub. That makes sponsoring difficult and also adding new maintainers.

It is an extremely widely used project and the MNF seems like the *perfect* place for it.

We’re not quite sure who at Google even owns this, any folks from Google here that could help?
 https://twitter.com/joshdholtz/status/1627362284623257601?s=46

**Comment by keith** [comment]
From the MNF side we'd love to have fastlane join and help out however we can!

**Reply by philipphofmann** [reply]
We, [Sentry](https://sentry.io/), yearly [donate](https://blog.sentry.io/2022/10/27/we-just-gave-260-028-dollars-to-open-source-maintainers) quite some money to open source, and we added it to our list of next fund round for 23, https://github.com/getsentry/team-ospo/issues/108.

**Reply by chadwhitacre** [reply]
> next fund round for 23

Getting to this now ... @joshdholtz @rogerluan looks like you are the two sponsorable maintainers of Fastlane? I've added you to our list of recipients in Thanks.dev (which can payout via GHS if you like). cc: @nehzata to support.

Thanks for your work!

**Reply by rogerluan** [reply]
Thank you thank you! Not sure how thanks.dev works, let me know if there's anything I can help you with!

**Comment by joshdholtz** [comment]
From the fastlane side, I'd love for it to be in MNF 🥰

**Comment by taquitos** [comment]
Maybe @google-ospo-team (@google-admin whoever is getting these notifications now) could help out here. This is a wonderful opportunity for Google to give back to oss while also helping to support a widely-used tool for both Android AND iOS.

**Comment by gsavit** [comment]
Thanks for kickstarting a discussion here @steipete, there have been rumblings but this is long overdue. Having chatted a bit with @joshdholtz earlier, I'm keen to throw my support behind making something happen here! Also, the team and I at Runway are very much prepared to offer whatever material support can help make this happen.

**Comment by mzsoliman** [comment]
It makes a lot of sense for Fastlane to be under MNF. It's a very critical piece of infrastructure for mobile developers and its success is crucial.
We at Instabug are happy to contribute to Fastlane as well both dedicated development resources as well as donations. And I'm pretty sure we can get others interested once this is under MNF.

**Comment by minuscorp** [comment]
I'd like to be part of MNF as well as @joshdholtz, I'd like to keep pushing fastlane for Swift and core issues as well in the spare time I can have.

**Reply by joshdholtz** [reply]
Oh, 100%! That current core _fastlane_ team members would still be the same (if they want of course 😉)

**Comment by jeroenleenarts** [comment]
If all else fails... there is always the option of a hard fork. Much rather see a transfer of IP though.

**Comment by donnie-jp** [comment]
+1

fastlane has been a really helpful tool for mobile devops tasks for many years and would be a great fit under MNF

**Comment by paulb777** [comment]
I'll follow up on this inside Google this week, but I don't know a lot about fastlane. What does it mean to transfer the IP for an MIT licensed project? What does Google need to do to enable a transfer to MNF?

**Reply by joshdholtz** [reply]
@paulb777 That would be amazing if you could 🥰  

And I think that @keith would probably have more info regarding the process of transferring IP/copyright for this with MNF... or at least would be able to find someone that would know 😊

**Reply by keith** [reply]
Yea the MNF works with some lawyers from the linux foundation for this. [Here](https://github.com/MobileNativeFoundation/foundation/blob/main/contribution-agreement.pdf) is the template contribution agreement, although technically things could change case by case. Not being a lawyer my assumption in this case is that the biggest piece is literally transferring ownership of the repos etc over.

**Reply by taquitos** [reply]
@paulb777 when I was at Google, I helped facilitate the transfer of "Google Science Journal" from Google to Arduino. You can see more here:
https://github.com/googlearchive/science-journal-ios
https://blog.arduino.cc/2020/08/05/the-science-journal-is-graduating-from-google-coming-to-arduino-this-fall/

This is the path I was hoping fastlane would take. Google just donates the project and the IP. Literally transferring the repo, the domains, the admin accounts, and helping to untangle what we tangled when Twitter sold us to Google in 2017 🤣.

To be clear, their most recent argument for not entertaining this is that "the best part about open source is that you can fork it!" -which, at best, totally ignores the community fragmentation and support issues- both monetary and engineering, along with aligning all external stakeholders on whether or not to use a fork, which fork is the "right", and then putting in the hours to update every CI provider.

**Reply by paulb777** [reply]
Thanks for the followups! There were several Googler conversations today about transferring fastlane. We have identified the right process and have begun working through the documentation, plans, and leadership approvals.

**Reply by joshdholtz** [reply]
Thanks so much for the update! Super happy to see a plan so soon 🥰

**Comment by jshchnz** [comment]
Fastlane is a crucial part of the development workflow. Of course us over at Emerge Tools would love to help! @joshdholtz let me know how we can be involved

**Comment by steipete** [comment]
Thanks for all your effort, folks! Just checking in here - do we have an update on the migration process?

**Reply by paulb777** [reply]
I'm sorry that I can't share the details, but we're making continued progress through the internal Google approval process.

**Reply by rogerluan** [reply]
A little over 3 weeks later, are there any new developments/progress @paulb777 ? 🙇 
Really the smallest tiniest little update you could give would fill our hearts with hope and warmth ❤️

**Reply by paulb777** [reply]
@rogerluan Still continuing to work through the approval process. There is continued progress, but many steps with busy people.

**Reply by rogerluan** [reply]
Thanks for the update @paulb777! 🙌

**Reply by bsrz** [reply]
@paulb777 apologies for being _that_ guy.
Are there any updates that you can share on the progress?
Thanks for all the efforts on this!

**Reply by paulb777** [reply]
Understood. This takes awhile. We're making progress. Lawyers are collaborating.

**Reply by rogerluan** [reply]
Hi @paulb777 👋 
Is this still moving forward? :)
Any light you could shed into how much progress has been made so far? Like, from 0 to 100%, where would we be? 🙇 

Thank you once again for all your effort on this! :bowtie:

**Reply by paulb777** [reply]
Hi @rogerluan There has been steady progress with several back and forth's between Google and MNF. I'm hopeful that we're close, but can't predict how long it will take to get all the necessary approvals.

**Reply by rogerluan** [reply]
Appreciate the update @paulb777! 🙌

**Comment by paulb777** [comment]
The fastlane transfer has now been approved by both Google and Mobile Native Foundation!

We've started the mechanics of the transfer by giving @keith ownership rights for https://github.com/fastlane and can begin working on migrating and removing the Google references in the repos.

I understand this took longer than we all hoped and I appreciate everyone's patience.

**Reply by keith** [reply]
Thanks so much for working on this Paul, and to everyone else behind the scenes at Google!

**Reply by paulb777** [reply]
Thanks to you Keith and everyone at MNF for making it happen! It's great to be able to work together to improve the developer eco-system.

**Reply by joshdholtz** [reply]
Thank you so much to both of you for working on this! Big thank you from me and the entire _fastlane_ core contributor team 🫶

**Reply by markltownsend** [reply]
I am grateful to see this happen. Fastlane is still the best automation tool for mobile that exists even after all these years.

**Reply by rogerluan** [reply]
Thank you to everyone involved! You all rock! 🏆❤️

**Reply by programming086** [reply]
🎉🎉🎉❤️

**Comment by schlagelk** [comment]
Hello, and thank you for the work to transfer ownership! Curious, when can we expect a new release of fastlane? It seems like a lot of people are still experience the issue described [here](https://github.com/fastlane/fastlane/issues/21475), which was fixed and merged [here](https://github.com/fastlane/fastlane/pull/21493) several weeks ago. We're seeing about 1/3 of our pipelines fail because of this and would like to avoid the overhead of installing fastlane from a gem. 

Thank you!

**Reply by rogerluan** [reply]
Hi @schlagelk 👋 

The fix to the issue in question has been released yesterday in https://github.com/fastlane/fastlane/releases/tag/2.217.0, thanks for the nudge 💪

**Reply by MikeChongCan** [reply]
same question here in April 2024

**Reply by rogerluan** [reply]
@imWildCat https://github.com/fastlane/fastlane/releases/tag/2.220.0

**Comment by yarneo** [comment]
Hey there! Can we get some updates on the progress of this move; and also get some maintenance activity of the existing Fastlane repo? The Xcode16 release broke Fastlane for us, and we opened a PR but getting crickets so far. I find the lack of support there somewhat concerning when relying on this project with large scale production products.

**Reply by MikeChongCan** [reply]
Let me share my own experience as well

https://github.com/fastlane/fastlane/pull/21052

for this PR I did get some feedbacks and really appreciate the time the reviewer spent 

however it is unclear to me about the rspec review feedback so I waited for some time
But the convo does not seem to be constructive anymore so I closed the PR

**Reply by TheMetalCode** [reply]
With respect to release cadence: I personally think it's ok for releases of features and non-critical bug fixes to occur monthly, maybe even quarterly. Given how critical Fastlane is for iOS/Android CI I think there's something to be said for stability and changes being somewhat less frequent than they might be for other projects.

However, when there are critical bug fixes that respond to underlying tooling changes (eg Xcode 16) and pinning to a prior Fastlane version isn't a viable workaround, that's when the ability to get a release out quickly would be appreciated, and that tends to be my biggest area of concern. I think @yarneo's PR (https://github.com/fastlane/fastlane/pull/22227) is a pretty good example of this case.

In fairness though: we did have a prior Xcode 16 breaking change on trainer (https://github.com/fastlane/fastlane/pull/22147) and it was released two days after merge. We do have recent experience of a quick release getting out when warranted.

**Reply by TheMetalCode** [reply]
@yarneo looks like your fix got picked up: https://github.com/fastlane/fastlane/releases/tag/2.223.0

**Reply by yarneo** [reply]
I can see! Thanks for the heads up 👍

**Comment by serso** [comment]
Hey, I've a security issue to report in fastlane. Where and how should I do that without disclosing it publicly?

**Comment by coopernetes** [comment]
Hi all, has there been any movement on "de-Googling" the project in light of https://github.com/MobileNativeFoundation/discussions/discussions/194#discussioncomment-7406759? Seems that this has stalled somewhat and we're really keen to see the project continue under the MNF's stewardship. We, like many other organizations, have a significant investment into Fastlane and wish to see it continue into the future. If needed, we can help contribute to those "tidying chores" if it can help to further this transition. Thanks!

**Reply by iBotPeaches** [reply]
I started "de-googling" the project. 

 - https://github.com/fastlane/fastlane.tools/pull/97 - docs
 - https://github.com/fastlane/code-of-conduct/pull/2 - code of conduct
 - https://github.com/fastlane/fastlane/pull/29747 - main repo
 - https://github.com/fastlane/codesigning.guide/pull/3 - code signing guide

**Reply by TomaszLizer** [reply]
OMG, we need this so much. Too many people depends on fastlane for it to die and recently it is not looking good.
Happy to assist with the movement and maintenance as I depend on it heavily in all my projects and still have love 😅
